packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.js

```diff
@@ -5,135 +5,127 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React, {useState, useRef, useCallback} from 'react';
-import classnames from 'classnames';
-
+import React, {useState, useCallback} from 'react';
+import {createPortal} from 'react-dom';
 import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
 import {useHistory} from '@docusaurus/router';
-import useSearchQuery from '@theme/hooks/useSearchQuery';
+import Link from '@docusaurus/Link';
+import Head from '@docusaurus/Head';
+import {DocSearchButton, useDocSearchKeyboardEvents} from '@docsearch/react';
 
-import styles from './styles.module.css';
+let DocSearchModal = null;
 
-const Search = ({handleSearchBarToggle, isSearchBarExpanded}) => {
-  const [algoliaLoaded, setAlgoliaLoaded] = useState(false);
-  const searchBarRef = useRef(null);
-  const {siteConfig = {}} = useDocusaurusContext();
-  const {
-    themeConfig: {algolia},
-  } = siteConfig;
+function Hit({hit, children}) {
+  return <Link to={hit.url}>{children}</Link>;
+}
+
+function ResultsFooter({state}) {
+  return (
+    <Link to={`/search?q=${state.query}`}>
```

Comment:
Use `navigateToSearchPage` from `useSearchQuery` hook to generate search link.

Reply:
How would you use it on a link element to keep it accessible? (allowing <kbd>Ctrl</kbd> click, etc.)

Reply:
I meant, we need to generate a search URL in `useSearchQuery` hook. Since `navigateToSearchPage` is not used anywhere else, we can rework it to something like `getSearchPage`.

Reply:
Gotcha: 032d937aa9c238c8ae8a28b207c9d19127c5e2f1.